### ISSUE TITLE
Save function missing 'overwrite' argument

### ISSUE THREAD

**Issue by bitbandit01** [barryvdh/laravel-snappy#33]
Hi,
First, thanks for making this wrapper, it has been of great help to me.
However, I have a small problem with the save function.   I need it to be able to overwrite the file if it already exists on the filesystem and this is possible with Snappy by passing the $overwrite = True option to the 'generate' function.  Unfortunately, this option is not included in your save function and so it throws an error if the file already exists.
I have fixed it on my own system by amending your function to the following:

    public function save($filename, $overwrite = false)
    {

        if ($this->html)
        {
            $this->snappy->generateFromHtml($this->html, $filename, $this->options, $overwrite);
        }
        elseif ($this->file)
        {
            $this->snappy->generate($this->file, $filename, $this->options, $overwrite);
        }

        return $this;
    }

I hope you will consider patching this for your next release as I think it is a handy function to have.
Sorry if this was not the correct way to submit this, but Im not all that familiar with github and patching.

**Comment by barryvdh** [comment]
I guess this would be useful. Not sure what the default would be for $overwrite. Perhaps it would be logical to overwrite by default?

**Comment by bitbandit01** [comment]
In Snappy itself, the default is false (GeneratorInterface.php Line 23).  I think it might actually be nicer to default it to false and have to explicitly include the overwrite argument when you want it to do so.  It seems safer that way as you might not be immediately aware that it will overwrite without looking at the code and so this could cause problems if files are overwriting when you didnt expect them to.

Also, if you happen to be digging into the code in the Snappy repository to try and figure things out (as I was when I discovered this option) you would probably expect the functionality to be consistent in the wrapper.

**Comment by barryvdh** [comment]
Agreed. If you want you can submit a Pull Request. You can do that with the Github webinterface, just edit the file: https://github.com/barryvdh/laravel-snappy/blob/master/src/Barryvdh/Snappy/PdfWrapper.php#L164 and follow the steps to create a PR.

**Comment by slickorange** [comment]
Is this going to be part of the official version?

**Comment by zlorkovic** [comment]
Yeah, also interested, will this be in official version soon? Very handy option.

**Comment by Jolora** [comment]
Yes :+1: - this would be a great little option in the official version. Thanks :)

**Comment by tanigami** [comment]
+1

**Comment by shaneparsons** [comment]
+1

**Comment by raffe90** [comment]
+1

**Comment by shaneparsons** [comment]
Any idea as to when this is going to be implemented?
Is there a current way to force a file overwrite?

**Comment by barryvdh** [comment]
What is the problem, you just need a new tag? Or do I need to change something still?

**Comment by shaneparsons** [comment]
Is there a way to implement overwriting within the save function? 
e.g. `$pdf->save($pdf);`

**Comment by jibelicious** [comment]
From what i can see, this is not in the code, 
At least not here ( https://github.com/barryvdh/laravel-snappy/blob/master/src/PdfWrapper.php )
I created a pull request ( https://github.com/barryvdh/laravel-snappy/pull/76 )
this would allow $pdf->save($pdf, $overwrite = true);

**Comment by PrateekSasanker** [comment]
Hi
While trying to save and overwrite pdf with $pdf->save($pdf, $overwrite = true); gives error "Could not delete already existing output file". How do I overwrite the existing file?

**Comment by Mr-Anonymous** [comment]
I think `$pdf->save($pdf, $overwrite = true);` is not right. It should be `$pdf->save($pdf, true);`
